src/common/types.d.ts

```diff
@@ -107,3 +150,62 @@ export interface GlobalState {
 export interface SocketState {
   [x: string]: any  // TODO Expose more field types as we need them
 }
+
+/* Game state subcomponents */
+
+export interface PlayerInGameState {
+  name: PlayerColor
+  robotsOnBoard: {
+    [hexId: string]: _Object
+  }
+  [x: string]: any  // TODO Expose more field types as we need them
+}
+
+interface _Object { // tslint:disable-line:class-name
```

Comment:
I'm not allowed to inherit from a type named `Object` because it's sort of a reserved word in Typescript. So I'm doing something kinda dumb and sneaky – calling the interface `_Object` internally but exporting it as `Object` below so I can still reference `Object`s in other files.